src/SourceBuild/content/eng/tools/BinaryToolKit/BinaryTool/Driver.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Logging;
+
+namespace BinaryTool;
+
+public static class Driver
+{
+    // Directory to scan for binaries: required
+    public static string TargetDirectory { get; set; } = string.Empty;
+
+    // Directory to output reports: required
+    public static string OutputReportDirectory { get; set; } = string.Empty;
+
+    // File for the baseline of allowed retainable binaries: optional
+    public static string AllowedBinariesKeepFile { get; set; } = string.Empty;
+
+    // File for the baseline of allowed removeable binaries: optional
+    public static string AllowedBinariesRemoveFile { get; set; } = string.Empty;
+
+    // Output files
+    public static string DetectedBinariesFile { get; private set; } = string.Empty;
+    public static string UpdatedAllowedBinariesKeepFile { get; private set; } = string.Empty;
+    public static string UpdatedAllowedBinariesRemoveFile { get; private set; } = string.Empty;
+    public static string NewBinariesFile { get; private set; } = string.Empty;
+    public static string RemovedBinariesFile { get; private set; } = string.Empty;
```

Comment:
Rather than relying on the caller to have to discover which of the static properties are relevant to output versus input, it'd be better for the `Execute` method to return this information as a return value.

Reply:
Actually, looking further into how these fields are set below, I'm not understanding why the `Driver` class is even providing this state. It doesn't look like any callers to `Driver.Execute` actually use this state. It's only the internal implementation within the flow of `Driver.Execute` that actually uses them. And that's because there are other static classes defined, like `RemoveBinaries`, that directly access that state. A better pattern would be to get rid of those classes like `RemoveBinaries` altogether, and move them into `Driver`. Then there is no state that needs to be exposed; it can be just kept internal. This can all be made simpler, IMO, if `Driver` was changed to be a non-static class and have its constructor compute these file paths and store them as fields.